packages/features/ee/teams/components/MemberListItem.tsx

```diff
@@ -23,93 +29,208 @@ import {
-        })),
-      };
+export type State = {
+  deleteMember: Payload;
+  impersonateMember: Payload;
+  inviteMember: Payload;
+  editSheet: Payload;
+  teamAvailability: Payload;
+  inviteLinkSetting: Payload;
+};
+
+export type Action =
+  | {
+      type:
+        | "SET_DELETE_ID"
+        | "SET_IMPERSONATE_ID"
+        | "INVITE_MEMBER"
+        | "EDIT_USER_SHEET"
+        | "TEAM_AVAILABILITY"
+        | "INVITE_LINK_SETTING";
+      payload: Payload;
     }
-  );
+  | {
+      type: "CLOSE_MODAL";
+    };
+
+const initialState: State = {
+  deleteMember: {
+    showModal: false,
+  },
+  impersonateMember: {
+    showModal: false,
+  },
+  inviteMember: {
+    showModal: false,
+  },
+  editSheet: {
+    showModal: false,
+  },
+  teamAvailability: {
+    showModal: false,
+  },
+  inviteLinkSetting: {
+    showModal: false,
+  },
 };
 
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case "SET_DELETE_ID":
+      return { ...state, deleteMember: action.payload };
+    case "SET_IMPERSONATE_ID":
+      return { ...state, impersonateMember: action.payload };
+    case "INVITE_MEMBER":
+      return { ...state, inviteMember: action.payload };
+    case "EDIT_USER_SHEET":
+      return { ...state, editSheet: action.payload };
+    case "INVITE_LINK_SETTING":
+      return { ...state, inviteLinkSetting: action.payload };
+    case "TEAM_AVAILABILITY":
+      return { ...state, teamAvailability: action.payload };
+    case "CLOSE_MODAL":
+      return {
+        ...state,
+        deleteMember: { showModal: false },
+        impersonateMember: { showModal: false },
+        inviteMember: { showModal: false },
+        editSheet: { showModal: false },
+        teamAvailability: { showModal: false },
+        inviteLinkSetting: { showModal: false },
+      };
+    default:
+      return state;
+  }
+}
+
 export default function MemberListItem(props: Props) {
   const { t, i18n } = useLocale();
-
+  const { data: session } = useSession();
   const utils = trpc.useUtils();
-  const [showChangeMemberRoleModal, setShowChangeMemberRoleModal] = useState(false);
-  const [showTeamAvailabilityModal, setShowTeamAvailabilityModal] = useState(false);
-  const [showImpersonateModal, setShowImpersonateModal] = useState(false);
-  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [searchTerm, setSearchTerm] = useState("");
+  const debouncedSearchTerm = useDebounce(searchTerm, 500);
+  const [connectedApps, setConnectedApps] = useState<Record<number, ConnectedAppsType[]>>({});
+  const [userIds, setUserIds] = useState<number[]>([]);
+
+  const { data: getUserConnectedApps } = trpc.viewer.teams.getUserConnectedApps.useQuery(
+    { userIds, teamId: props.team.id },
+    { enabled: !!userIds.length }
+  );
+
```

Comment:
if we are displaying connected apps only in edit modal then we should not fetch it before the modal is clicked.

Reply:
Yeah, make sense